File: Src/App/zabotalaboratory/zabotalaboratory.Analyses.Database/Entities/LaboratoryAnalyses.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using zabotalaboratory.Common.Datamodel.Abstractions;

namespace zabotalaboratory.Analyses.Database.Entities
{
    public class LaboratoryAnalyses : IValidatable
    {
        public virtual int Id { get; set; }

        [Required]
        [MaxLength(32)]
        public virtual string LastName { get; set; }

        [Required]
        [MaxLength(32)]
        public virtual string FirstName { get; set; }

        [Required]
        [MaxLength(32)]
        public virtual string PatronymicName { get; set; }

        [Required]
        public virtual DateTimeOffset DateOfBirth { get; set; }

        [ForeignKey(nameof(Gender))]
        public virtual int GenderId { get; set; }

        [Required]
        public virtual Gender Gender { get; set; }

        [Required]
        [ForeignKey(nameof(Clinic))]
        public virtual int ClinicId { get; set; }

        [Required]
        public virtual Clinics Clinic { get; set; }

        [Required]
        public virtual DateTimeOffset PickUpDate { get; set; }

        public virtual string Doctor { get; set; }

        public virtual List<Talons> Talons { get; set; }

        public virtual MedicalRecord MedicalRecord { get; set; }

        public virtual bool IsValid { get; set; }
    }
}

File: Src/App/zabotalaboratory_client/src/app/shared/models/analyses/laboratory-analyses.ts
import { Clinic } from './clinic';
import { Talons } from './talons';
import {Gender} from "./gender";
import {MedicalRecord} from "./medical-record";

export class LaboratoryAnalyses{
  public id: number;
  public lastName: string;
  public firstName: string;
  public patronymicName: string;
  public fullName: string;
  public dateOfBirth: Date;
  public gender: Gender;
  public clinicId: number;
  public clinic: Clinic;
  public pickUpDate: Date;
  public talons: Talons[];
  public doctor: string;
  public medicalRecord: MedicalRecord;
  public isValid: boolean;
}
